{
	"identifier": "CVE-2022-40899",
	"description": "An issue discovered in Python Charmers Future 0.18.2 and earlier allows remote attackers to cause a denial of service via crafted Set-Cookie header from malicious web server.",
	"type": "none",
	"severity": "medium",
	"state": "published",
	"datetime": "2022-12-22T22:00:00Z",
	"is_edited": false,
	"hardware": [],
	"software": [
		{
			"name": "python-future",
			"version": "any",
			"architecture": "any",
			"manager": "apt",
			"vendor": "debian-bullseye"
		},
		{
			"name": "python-future",
			"version": "any",
			"architecture": "any",
			"manager": "apt",
			"vendor": "debian-buster"
		},
		{
			"name": "python-future",
			"version": "any",
			"architecture": "any",
			"manager": "apt",
			"vendor": "debian-bookworm"
		},
		{
			"name": "python-future",
			"version": "any",
			"architecture": "any",
			"manager": "apt",
			"vendor": "debian-sid"
		},
		{
			"name": "python-future",
			"version": "any",
			"architecture": "any",
			"manager": "apt",
			"vendor": "ubuntu-trusty"
		},
		{
			"name": "python-future",
			"version": "\u003c 0.15.2-1ubuntu0.1~esm1",
			"architecture": "any",
			"manager": "apt",
			"vendor": "ubuntu-xenial"
		},
		{
			"name": "python-future",
			"version": "\u003c 0.15.2-4ubuntu2.1",
			"architecture": "any",
			"manager": "apt",
			"vendor": "ubuntu-bionic"
		},
		{
			"name": "python-future",
			"version": "\u003c 0.18.2-2ubuntu0.1",
			"architecture": "any",
			"manager": "apt",
			"vendor": "ubuntu-focal"
		},
		{
			"name": "python-future",
			"version": "\u003c 0.18.2-5ubuntu0.1",
			"architecture": "any",
			"manager": "apt",
			"vendor": "ubuntu-jammy"
		},
		{
			"name": "python-future",
			"version": "\u003c 0.18.2-6ubuntu0.1",
			"architecture": "any",
			"manager": "apt",
			"vendor": "ubuntu-kinetic"
		}
	],
	"products": [
		{
			"name": "python-future",
			"version": "\u003c= 0.18.2",
			"architecture": "any",
			"vendor": "pythoncharmers",
			"type": "software",
			"is_edited": false,
			"aliases": []
		}
	],
	"references": [
		"https://github.com/PythonCharmers/python-future/blob/master/src/future/backports/http/cookiejar.py#L215",
		"https://github.com/python/cpython/pull/17157",
		"https://pypi.org/project/future/",
		"https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/",
		"https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-40899",
		"https://github.com/PythonCharmers/python-future/commit/c91d70b34ef0402aef3e9d04364ba98509dca76f",
		"https://ubuntu.com/security/notices/USN-5833-1"
	],
	"weaknesses": []
}